### DISCUSSION TITLE
Feasibility/request for virtio in the packet driver interface

### DISCUSSION THREAD

**Discussion by jharrison03** [dosemu2/dosemu2#1744] [Ideas]
KVM is working pretty good, I think at this point, and it might be beneficial to look at virtio in the packet driver interface.

**Comment by stsp** [comment]
No it won't be beneficial.
virtio is a guest-aware transport.
Unless there is a virtio-net packet
driver for DOS, nothing to look into.

**Comment by stsp** [comment]
Are you aware that we already
use slirp by default? VDE is no
longer needed.

**Reply by jharrison03** [reply]
I don't understand what SLIRP has to do with this?  DOSemu pegs a core to 100% during network traffic vs. a Windows machine with virtio-net drivers which uses baically zero ( 0% ).

**Reply by stsp** [reply]
Use int28 in your program to
give up CPU cycles.

**Reply by jharrison03** [reply]
So I should call int28 while I'm doing network transfers.  Precisely how often?

**Reply by stsp** [reply]
Packet driver calls your receiver
routine. That means you can send
the data and then do int28 as much
as you want, while waiting for pkt
driver to call your receiver.

**Comment by jharrison03** [comment]
I don't use IP networking, but rather IPX where SLIRP isn't relevant, and IPX has been removed from the kernel - so I have to do this client-side, either with VDE or TUN, and there doesn't seem to be any advantage WRT to TUN over VDE.

**Comment by stsp** [comment]
> and there doesn't seem to be any advantage WRT to TUN over VDE.

There is.
VDE in that case is just an
unnecessary, thick layer of
unmaintained code, that, at
the end, leads you to the
same TAP device.
I added vde support only
because there is a vdeslirp
plugin. But in your case its
not used.
